src/librustc_span/lib.rs

```diff
@@ -888,6 +892,36 @@ impl ExternalSource {
 #[derive(Debug)]
 pub struct OffsetOverflowError;
 
+#[derive(Copy, Clone, RustcEncodable, RustcDecodable)]
+pub enum SourceFileChecksumKind {
+    Md5,
+    Sha1,
+}
+
+/// The hash of the on-disk source file used for debug info.
+#[derive(Clone, RustcEncodable, RustcDecodable)]
+pub struct SourceFileChecksum {
+    pub kind: SourceFileChecksumKind,
+    value: Vec<u8>,
+}
+
+impl SourceFileChecksum {
+    pub fn bytes(&self) -> &[u8] {
```

Comment:
Nit: This should probably `as_bytes` which is the standard borrow to borrow method naming convention. [See String as an example.](https://doc.rust-lang.org/std/string/struct.String.html#method.as_bytes)